main.py

```diff
@@ -90,7 +90,7 @@ def check_pre_requisites_tesseract():
         return True
 
 
-def main(input_path, output_path):
+def main(input_path, output_path, no_output_directory=False):
```

Comment:
Why do you need a new parameter to detect is output directory was provided or not?
You can pass `output_path` as `None` and use that as your flag.